component/azstorage/block_blob.go

```diff
@@ -404,13 +391,25 @@ func (bb *BlockBlob) RenameDirectory(source string, target string) error {
 		}
 	}
 
-	err := bb.RenameFile(source, target)
-	// check if the marker blob for source directory does not exist but
-	// blobs were present in it, which were renamed earlier
-	if err == syscall.ENOENT && srcDirPresent {
-		err = nil
+	// To rename source marker blob check its properties before calling rename on it.
+	blobClient := bb.Container.NewBlockBlobClient(filepath.Join(bb.Config.prefixPath, source))
+	_, err := blobClient.GetProperties(context.Background(), &blob.GetPropertiesOptions{
+		CPKInfo: bb.blobCPKOpt,
+	})
+	if err != nil {
+		serr := storeBlobErrToErr(err)
+		if serr == ErrFileNotFound && srcDirPresent {
+			return nil
+		} else if serr == ErrFileNotFound {
+			log.Err("BlockBlob::RenameDirectory : %s marker blob does not exist and Src Directory doesn't Exist", source)
+			return syscall.ENOENT
+		} else {
+			log.Err("BlockBlob::RenameDirectory : Failed to get source directory marker blob properties for %s [%s]", source, err.Error())
+			return err
+		}
```

Comment:
> 
>if serr == ErrFileNotFound && srcDirPresent {
			return nil
		} else if serr == ErrFileNotFound {
			log.Err("BlockBlob::RenameDirectory : %s marker blob does not exist and Src Directory doesn't Exist", source)
			return syscall.ENOENT
		} else {
			log.Err("BlockBlob::RenameDirectory : Failed to get source directory marker blob properties for %s [%s]", source, err.Error())
			return err
		}
>


Simplified code can be
`
if serr == ErrFileNotFound {
		if srcDirPresent {
			return nil
		}
		log.Err("BlockBlob::RenameDirectory: Marker blob %s and source directory do not exist.", source)
		return syscall.ENOENT
	}

	// Other errors when getting blob properties
	log.Err("BlockBlob::RenameDirectory: Failed to get properties for marker blob %s: [%s]", source, err.Error())
	return err
`

Reply:
Modified the Code